maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/NewGradleJSFModuleProjectProvider.java

```diff
@@ -0,0 +1,224 @@
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.m2e.core.internal.MavenPluginActivator;
+import org.eclipse.m2e.core.project.ProjectImportConfiguration;
+import org.eclipse.m2e.core.project.ResolverConfiguration;
+import org.eclipse.sapphire.platform.PathBridge;
+
+/**
+ * @author Simon Jiang
+ */
+
+@SuppressWarnings( "restriction" )
+public class NewGradleJSFModuleProjectProvider extends NewMavenJSFModuleProjectProvider
+{
+
+    @Override
+    public IStatus createNewProject( NewLiferayJSFModuleProjectOp op, IProgressMonitor monitor ) throws CoreException
+    {
+        IStatus retval = Status.OK_STATUS;;
+
+        try
+        {
+            final String projectName = op.getProjectName().content();
+
+            IPath location = PathBridge.create( op.getLocation().content() );
+
+            // for location we should use the parent location
+            if( location.lastSegment().equals( projectName ) )
+            {
+                // use parent dir since maven archetype will generate new dir under this location
+                location = location.removeLastSegments( 1 );
+            }
+
+            final String groupId = op.getGroupId().content();
+            final String artifactId = op.getProjectName().content();
+            final String version = op.getArtifactVersion().content();
+
+            final String archetypeArtifactId = op.getArchetype().content();
+
+            final Archetype archetype = new Archetype();
+
+            final String[] gav = archetypeArtifactId.split( ":" );
+
+            final String archetypeVersion = gav[gav.length - 1];
+
+            archetype.setGroupId( gav[0] );
+            archetype.setArtifactId( gav[1] );
+            archetype.setVersion( archetypeVersion );
+
+            Artifact mavenArchtype = AetherUtil.getMavenArchtype(
+                archetype.getGroupId(), archetype.getArtifactId(), archetype.getVersion() );
+
+            final ResolverConfiguration resolverConfig = new ResolverConfiguration();
+            ProjectImportConfiguration configuration = new ProjectImportConfiguration( resolverConfig );
+            final Properties properties = new Properties();
+
+            IPath projectLocation = createArchetypeProject(
+                mavenArchtype, location, archetype, groupId, artifactId, version, "com.example", properties,
+                configuration, monitor );
+
+            IPath buildGradlePath = projectLocation.append( "build.gradle" );
+
+            if( buildGradlePath.toFile().exists() )
+            {
+                try
+                {
+                    boolean hasLiferayWorkspace = LiferayWorkspaceUtil.hasLiferayWorkspace();
```

Comment:
our current solution is we will generate project location when user type projectName, so its no problem when we only have one liferay workspace, we can easily to set a project stored in liferay workspace.  If the eclipse has more than one workspace, its not enough that we only to get to know the project location is in one liferay workspace folder..